core/embed/trezorhal/stm32f4/supervise.h

```diff
@@ -58,10 +60,22 @@ static inline void svc_shutdown(void) {
     shutdown_privileged();
   }
 }
+
 static inline void svc_reboot_to_bootloader(void) {
   if (is_mode_unprivileged() && !is_mode_handler()) {
     __asm__ __volatile__("svc %0" ::"i"(SVC_REBOOT_TO_BOOTLOADER) : "memory");
   } else {
     reboot_to_bootloader();
   }
 }
+
+static inline void svc_reboot_copy_image_header(const uint8_t *image_address) {
+  if (is_mode_unprivileged() && !is_mode_handler()) {
+    register const uint8_t *r0 __asm__("r0") = image_address;
+    __asm__ __volatile__("svc %0" ::"i"(SVC_REBOOT_COPY_IMAGE_HEADER), "r"(r0)
+                         : "memory");
+  } else {
+    copy_image_header_for_bootloader(image_address);
```

Comment:
i know this does not originate in this PR: but perhaps we should call `ensure_compatible_settings` here too? i don't think that the else branch will be called anywhere today, but lets avoid the potential problem in future (also valid for the `svc_reboot_to_bootloader` function above). Also `mpu_config_bootloader` is missing here in comparison with the SVC call

Reply:
Fixed as we discussed.